tests/ignite/handlers/test_ema_handler.py

```diff
@@ -0,0 +1,296 @@
+import os
+from typing import Any, Callable, Union
+
+import pytest
+import torch
+import torch.nn as nn
+from torch.nn.parallel import DataParallel, DistributedDataParallel
+from torch.utils.data import DataLoader, TensorDataset
+
+import ignite.distributed as idist
+from ignite.engine import Engine, Events
+from ignite.handlers import EMAHandler
+
+
+def _get_dummy_dataloader() -> DataLoader:
+    """Get a dummy data loader with length of 2"""
+    x = torch.randint(0, 10, (2, 2))
+    data_set = TensorDataset(x)
+    data_loader = DataLoader(data_set, batch_size=1)
+    return data_loader
+
+
+def _get_dummy_model() -> nn.Module:
+    model = nn.Linear(2, 1, bias=False)
+    model.weight.data.fill_(1)
+    return model
+
+
+def _get_dummy_step_fn(model: Union[nn.Module, DataParallel, DistributedDataParallel]) -> Callable:
+    """Get a dummy step function, given model is a (wrapper of) dummy model returned from _get_dummy_model"""
+
+    def step_fn(engine, batch):
+        """Increment the weight by 1 at each iteration"""
+        model.weight.data.add_(1)
+        return 0
+
+    return step_fn
+
+
+@pytest.mark.parametrize("momentum_end", [-1, 2])
+def test_ema_invalid_momentum_end(momentum_end):
+    with pytest.raises(ValueError):
```

Comment:
Usually, we also provide `match="part of error message to match"` argument to ensure which message we check.

Reply:
Done